blocks/footer/footer.js

```diff
@@ -1,21 +1,155 @@
-import { getMetadata } from '../../scripts/aem.js';
-import { loadFragment } from '../fragment/fragment.js';
+import {
+  decorateSections,
+  updateSectionsStatus,
+} from '../../scripts/aem.js';
+
+function onClick() {
+  const disclaimerDiv = document.querySelector('.panel-body');
+  const iconDiv = document.querySelector('.toggle-icon');
+  if (disclaimerDiv.classList.contains('slide-up')) {
+    // If the content is currently hidden, slide it down
+    disclaimerDiv.classList.remove('slide-up');
+    iconDiv.textContent = '-';
+  } else {
+    // If the content is currently visible, slide it up
+    disclaimerDiv.classList.add('slide-up');
+    iconDiv.textContent = '+';
+  }
+}
+
+function decorateFooterTop(block) {
+  const footerTop = block.querySelector('.footer-top');
+  const defaultWrapper = footerTop.querySelector('.default-content-wrapper');
+  const children = [...footerTop.querySelector('.default-content-wrapper').children];
+  defaultWrapper.innerHTML = '';
+  const rowElement = document.createElement('div');
+  defaultWrapper.appendChild(rowElement);
+
+  const col1 = document.createElement('div');
+  col1.classList.add('col-1');
+  col1.appendChild(children[0]);
+  const imageElement = document.createElement('img');
+  const anchorDiv = children[1].childNodes[0];
+  imageElement.src = anchorDiv.href;
+  imageElement.alt = 'ICICI Logo Footer';
+  col1.appendChild(imageElement);
+
+  const col2 = document.createElement('div');
+  col2.classList.add('col-2');
+  col2.appendChild(children[2]);
+  children[3].childNodes[0].classList.add('toll-free-number');
```

Comment:
I think we are making a few assumptions in code that the authors need to follow in the "footer top" section like placing the logo first followed by 2 phone numbers. This would break if authors let's say placed the phone numbers first and then the logo.
To simplify this for authors, I'd suggest you create a table in this section like

|||
|---|---|
| Logo | logo image|
| Toll Free Number | number|
| Toll Free Number | number|

Reply:
as satya suggested, making a table like this would help reading these values as block config key value pair

Reply:
agreed. working on it

Reply:
creating table would considered as block and try to get js and css for that block and produce error, is there any meta data which can avoid it ?

Reply:
done